### PR TITLE
Correctly detect powershell when minikube is called on windows through SSH

### LINKED ISSUE #10626: Failing test: TestForceSystemdFlag on Windows Docker
https://storage.googleapis.com/minikube-builds/logs/10603/fb9f08b/Docker_Windows.html#fail_TestForceSystemdFlag

### PR DESCRIPTION
**Pull request by hetong07**
Fix #10626 

Make the minikube correctly detect it is running in powershell.

**Before PR**:
```
PS > \minikube-windows-amd64.exe docker-env
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.101:2376"
export DOCKER_CERT_PATH="C:\Users\imwise\.minikube\certs"
export MINIKUBE_ACTIVE_DOCKERD="minikube"

# To point your shell to minikube's docker-daemon, run:
# eval $(minikube -p minikube docker-env)
```
**After PR**:
```
PS > .\minikube-windows-amd64.exe docker-env
$Env:DOCKER_TLS_VERIFY = "1"
$Env:DOCKER_HOST = "tcp://192.168.99.101:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\imwise\.minikube\certs"
$Env:MINIKUBE_ACTIVE_DOCKERD = "minikube"
# To point your shell to minikube's docker-daemon, run:
# & minikube -p minikube docker-env | Invoke-Expression

```

### CODE REVIEW

File: cmd/minikube/cmd/docker-env.go [hunk 1]
```diff
@@ -245,6 +245,7 @@ var dockerEnvCmd = &cobra.Command{
 		shl := shell.ForceShell
 		if shl == "" {
 			shl, err = shell.Detect()
+			shl = strings.Trim("powershell.exe", ".exe")
```
Comment (ilya-zuyev) [inline cmd/minikube/cmd/docker-env.go]: The result of `strings.Trim("powershell.exe", ".exe")` is always `powershell`. Here we effectively just set shl = "powershell" and ignore the result of `shell.Detect()`. Not sure it's that we really need here.

Comment (medyagh) [reply cmd/minikube/cmd/docker-env.go]: @hetong07 this seems like a mistake to me, I wonder if u meant to do something else ? simmilar to ilya I suggest we fix the shell.Detect() logic so it does the right thing

File: cmd/minikube/cmd/docker-env.go [hunk 2]
```diff
@@ -245,6 +246,7 @@ var dockerEnvCmd = &cobra.Command{
 		shl := shell.ForceShell
 		if shl == "" {
 			shl, err = shell.Detect()
+			shl = strings.TrimSuffix(shl, filepath.Ext(shl))
```
Comment (ilya-zuyev) [inline cmd/minikube/cmd/docker-env.go:249]: Would be a little bit better to first check the error and then if it's nil trim the '.exe'

File: go.mod [hunk 3]
```diff
@@ -4,7 +4,7 @@ go 1.15
 
 require (
 	cloud.google.com/go/storage v1.13.0
-	contrib.go.opencensus.io/exporter/stackdriver v0.12.1
+	contrib.go.opencensus.io/exporter/stackdriver v0.13.5
```
Comment (ilya-zuyev) [inline go.mod:7]: If this PR can work with old versions of dependencies, it's better to keep it and update go.[mod|sum] in a separate PR

Comment (hetong07) [reply go.mod]: Directly copied and replace the go.mod and go.sum file from minikube master branch will resolve this issue.

File: cmd/minikube/cmd/docker-env.go (hunk 1 above)
Comment (hetong07) [reply cmd/minikube/cmd/docker-env.go]: @medyagh and @ilya-zuyev Sorry I mistakenly upload the experimental changes here, and have fixed it in subsequent CL, do I need to make a --force push to overwrite this commit?

Comment (hetong07) [reply cmd/minikube/cmd/docker-env.go]: @medyagh and @ilya-zuyev do a check and find my branch has divert from the minikube HEAD, so I will close issue and create another PR to solve this problem.

**Pull request closed without merging**